runtime/compiler/trj9/il/J9SymbolReference.cpp

```diff
@@ -258,14 +258,16 @@ SymbolReference::getTypeSignature(int32_t & len, TR_AllocationKind allocKind, bo
                   !comp->getOption(TR_AOT) &&
                    (type == TR::Address)))
                  {
-                 void * p = symbol->getStaticAddress();
+                 uintptrj_t objectStaticAddress = (uintptrj_t)symbol->getStaticAddress();
                  TR::VMAccessCriticalSection vmAccessCriticalSection(comp->fej9(),
                                                                       TR::VMAccessCriticalSection::tryToAcquireVMAccess,
                                                                       comp);
-                 if ((*(void **)p != 0) &&
+
+                 uintptrj_t objectRef = comp->fej9()->getStaticReferenceFieldAtAddress(comp, objectStaticAddress, (TR::SymbolReference *)this);
```

Comment:
VM access might not be held here IIUC

Reply:
Yeah you're right. The reason I left it as is was because:

1. If we failed to acquire vm access, the next condition in the `if` will fail, so the object getting moved twice doesn't matter.
2. if we acquired vm access, the code executes as expected.

However, now that I think of it, it might not be ok to execute the read barrier without VM Access, so it's probably for the best to switch the order of the conditions.

Reply:
> it might not be ok to execute the read barrier without VM Access

Yeah, I'm not sure whether that's safe either, but I think `getStaticReferenceFieldAtAddress` wants to require VM access